tests/asgi/tests.py

```diff
@@ -104,7 +105,8 @@ async def test_file_response(self):
             "django.contrib.staticfiles.finders.FileSystemFinder",
         ],
     )
-    async def test_static_file_response(self):
+    @mock.patch("tempfile.SpooledTemporaryFile.close")
+    async def test_static_file_response(self, close_stream_mock):
```

Comment:
The `patch` here means that the underlying `close()` isn't performed, resulting in the... 

```
ResourceWarning: Unclosed file <tempfile.SpooledTemporaryFile object at 0x...>
```

... on Python 3.11. 

I'm not sure if it's worth checking — If it's not called, we'll get the `ResourceWarning`. 🤔

@lundberg Can you test against the 3.11 beta in order to check this yourself?

Reply:
I hear you 😉 

I'm thinking dropping the `patch`, and instead try and see if I can assert that there's no `ResourceWarning` warned. 

Should work for <3.11 since there's no warning at all, and for 3.11 give us validation that the stream is closed 🤞.

Agree @carltongibson ?

Reply:
@lundberg The test runner already [raises `ResourceWarning` as an error](https://github.com/django/django/blob/49b470b9187b6be60e573fed08c8f4a87f133750/tests/runtests.py#L50) -- so  that should be enough... — it's implicit but sufficient (since some of these don't get raised until shutdown).

Happy to look at a warnings filter context manager version, if that's what you've got in mind, but I'm not sure it's needed. 🤔